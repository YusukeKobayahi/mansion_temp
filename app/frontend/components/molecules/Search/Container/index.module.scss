@import "~/styles/variables";
@import "~/styles/utils";

.container {
  width: 100%;
  border: 1px solid #e5e5e5;
  border-radius: 4px;
  background-color: #fff;
}

.title {
  display: flex;
  align-items: center;
  width: 100%;
  border-radius: 4px 4px 0 0;
  background-color: #f4f4f4;
  padding-left: 10px;

  @include screenMd {
    padding: 10px 16px;
    width: 100%;
    margin: 0 auto;
    border-radius: 0;
    box-sizing: border-box;
  }

  p {
    line-height: 40px;
    font-size: 16px;

    @include screenMd {
      line-height: unset;
      padding-left: 10px;
    }
  }
}

.body {
  display: block;
  padding: 20px;

  @include screenMd {
    height: 100%;
    overflow-y: auto;
    max-height: calc(100vh - 85px);
    display: none;
    padding: 15px 16px 50px;
    box-sizing: border-box;
  }
}

.toggler {
  margin-left: auto;
  cursor: pointer;
  background-color: transparent;
  align-self: center;

  span {
    display: block;
    width: 11px;
    height: 11px;
    border-right: 1px solid #000;
    border-bottom: 1px solid #000;
    transform-origin: center center;
  }
  .open {
    transform: translateY(30%) rotate(-135deg);
  }
  .close {
    transform: translateY(-30%) rotate(45deg);
  }
}

.visible {
  display: block;
}
.invisible {
  display: none;
}
