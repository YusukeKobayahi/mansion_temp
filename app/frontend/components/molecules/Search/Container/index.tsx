import Image from "next/image";
import classnames from "classnames";
import styles from "./index.module.scss";
import { ReactSetStateType } from "~/lib/types";
import ConditionalWrapper from "~/components/commons/ConditionalWrapper";
import Minimizer from "~/components/molecules/Search/Minimizer";

type ContainerProps = {
  children: React.ReactNode;
  title?: string;
  minimize?: boolean;
  isOpen?: boolean;
  withTitle?: boolean;
  setIsOpen?: ReactSetStateType<boolean>;
};

const Container: React.FC<ContainerProps> = ({
  title = "絞り込み条件",
  minimize = false,
  children,
  isOpen = false,
  withTitle = true,
  setIsOpen = () => {
    return;
  },
}: ContainerProps) => {
  const togglerClasses = classnames({
    [styles.invisible]: minimize,
    [styles.toggler]: true,
  });
  const bodyClasses = classnames({
    [styles.body]: true,
    [styles.visible]: isOpen || minimize,
  });
  return (
    <ConditionalWrapper
      condition={minimize}
      wrapper={(children) => (
        <Minimizer setIsOpen={setIsOpen} isOpen={isOpen}>
          {children}
        </Minimizer>
      )}
    >
      <section>
        <div className={styles.container}>
          {withTitle && (
            <>
              <div className={"pcOnly"}>
                <div className={styles.title}>
                  <p>{title}</p>
                </div>
              </div>
              <div className={"spOnly"}>
                <button
                  className={styles.title}
                  onClick={() => setIsOpen(!isOpen)}
                >
                  <Image
                    src={"/ms-library/images/option.svg"}
                    alt={"option"}
                    width={20}
                    height={20}
                  />
                  <p>{title}</p>
                  <div className={togglerClasses}>
                    <span
                      className={isOpen ? styles.open : styles.close}
                    ></span>
                  </div>
                </button>
              </div>
            </>
          )}
          <div className={bodyClasses}>{children}</div>
        </div>
      </section>
    </ConditionalWrapper>
  );
};

export default Container;
