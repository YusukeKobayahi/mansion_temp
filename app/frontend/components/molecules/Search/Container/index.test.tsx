import { render } from "@testing-library/react";
import Container from ".";

describe("components/molecules/Search/Container", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(
      <Container
        setIsOpen={() => {
          return;
        }}
        isOpen={true}
      >
        <span>button</span>
      </Container>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
