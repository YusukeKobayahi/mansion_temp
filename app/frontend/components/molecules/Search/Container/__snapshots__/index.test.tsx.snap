// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/Search/Container 正常に表示される 1`] = `
<DocumentFragment>
  <section>
    <div
      class="container"
    >
      <div
        class="pcOnly"
      >
        <div
          class="title"
        >
          <p>
            絞り込み条件
          </p>
        </div>
      </div>
      <div
        class="spOnly"
      >
        <button
          class="title"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAiIGhlaWdodD0iMjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="option"
              data-nimg="true"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <p>
            絞り込み条件
          </p>
          <div
            class="toggler"
          >
            <span
              class="open"
            />
          </div>
        </button>
      </div>
      <div
        class="body visible"
      >
        <span>
          button
        </span>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
